### ISSUE TITLE
Panic when cross join two empty Frame

### ISSUE THREAD

**Issue by gab23r** [pola-rs/polars#7245]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

polars panics when we try to cross join two empty Lazy/DataFrames  

### Reproducible example

```python
import polars as pl
a = pl.LazyFrame(schema={'a': pl.Int32})
b = pl.LazyFrame(schema={'b': pl.Int32})

_ = (
    a
    .join(b, how='cross')
    .collect()
)
# PanicException: called `Option::unwrap()` on a `None` value
```

### Expected behavior

An empty DataFrame with this schema {'a': pl.Int32, 'b': pl.Int32}

### Installed versions

<details>

```
---Version info---
Polars: 0.16.9
Index type: UInt32
Platform: Linux-3.10.0-862.el7.x86_64-x86_64-with-glibc2.17
Python: 3.10.9 | packaged by conda-forge | (main, Feb  2 2023, 20:20:04) [GCC 11.3.0]
---Optional dependencies---
pyarrow: 11.0.0
pandas: 1.5.3
numpy: 1.24.1
fsspec: 2023.1.0
connectorx: 0.3.1
xlsx2csv: <not installed>
deltalake: <not installed>
matplotlib: <not installed>
```

</details>

**Comment by mcrumiller** [comment]
Doesn't even have to be lazy:

```python
pl.DataFrame().join(pl.DataFrame(), how="cross")
```
```
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', D:\a\polars\polars\polars\polars-core\src\utils\mod.rs:620:34
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Projects\project-brr\.venv_brr\lib\site-packages\polars\utils.py", line 498, in wrapper
    return function(*args, **kwargs)
  File "C:\Projects\project-brr\.venv_brr\lib\site-packages\polars\internals\dataframe\frame.py", line 4489, in join
    self.lazy()
  File "C:\Projects\project-brr\.venv_brr\lib\site-packages\polars\internals\lazyframe\frame.py", line 1323, in collect
    return pli.wrap_df(ldf.collect())
pyo3_runtime.PanicException: called `Option::unwrap()` on a `None` value
```

Granted, a pretty ridiculous edge case, but probably worth fixing.

**Comment by ritchie46** [comment]
fixed by #7266
